delta/app/src/main/scala/ch/epfl/bluebrain/nexus/delta/routes/AclsRoutes.scala

```diff
@@ -0,0 +1,277 @@
+      case Path.Slash(Path.Segment(org, Path.Empty)) if org != any                                                => provide(Organization(Label.unsafe(org)))
+      case Path.Slash(Path.Segment(org, Path.Slash(Path.Segment(proj, Path.Empty)))) if org != any && proj != any =>
+        (for {
+          orgLabel  <- Label(org)
+          projLabel <- Label(proj)
+        } yield Project(orgLabel, projLabel)).fold(
+          err => reject(validationRejection(err.getMessage)),
+          provide(_)
+        )
+      case _                                                                                                      => reject
+
+    }
+
+  private def extractAclAddressFilter: Directive1[AclAddressFilter] =
+    (extractUnmatchedPath & parameter("ancestors" ? false)).tflatMap { case (path, ancestors) =>
+      path match {
+        case Path.Slash(Path.Segment(`any`, Path.Empty))                                  => provide(AnyOrganization(ancestors))
+        case Path.Slash(Path.Segment(`any`, Path.Slash(Path.Segment(`any`, Path.Empty)))) =>
+          provide(AnyOrganizationAnyProject(ancestors))
+        case Path.Slash(Path.Segment(org, Path.Slash(Path.Segment(`any`, Path.Empty))))   =>
+          Label(org).fold(
+            err => reject(validationRejection(err.getMessage)),
+            label => provide(AnyProject(label, ancestors).asInstanceOf[AclAddressFilter])
+          )
+        case _                                                                            => reject
+      }
+    }
+
+  def routes: Route = baseUriPrefix(baseUri.prefix) {
+    pathPrefix("acls") {
+      extractCaller { implicit caller =>
+        concat(
+          extractAclAddress { address =>
+            extractSubject { implicit subject =>
+              parameter("rev" ? 0L) { rev =>
+                concat(
+                  (put & entity(as[AclInput])) { aclEntity =>
+                    authorizeFor(address, aclsWrite).apply {
+                      val status = if (rev == 0L) Created else OK
+                      completeIO(status, acls.replace(address, aclEntity.toAcl, rev).map(_.void))
+                    }
+                  },
+                  (patch & entity(as[PatchAcl])) { aclPatch =>
+                    authorizeFor(address, aclsWrite).apply {
+                      aclPatch match {
+                        case AppendAcl(acl)   =>
+                          completeIO(acls.append(address, acl, rev).map(_.void))
+                        case SubtractAcl(acl) =>
+                          completeIO(acls.subtract(address, acl, rev).map(_.void))
+                      }
+                    }
+                  },
+                  delete {
+                    authorizeFor(address, aclsWrite).apply {
+                      completeIO(OK, acls.delete(address, rev).map(_.void))
+                    }
+                  },
+                  (get & parameter("self" ? true)) {
+                    case true  =>
```

Comment:
Why would you need `acls/read` permissions only to read acls when the self=false and not in this case? If that was previously the logic, it seems flawed.

Reply:
it seems that currently, you can always read your own (`self`) ACLs.